mne/decoding/transformer.py

```diff
@@ -245,6 +245,97 @@ def inverse_transform(self, X, y=None):
         return X.reshape(-1, self.n_channels, self.n_times)
 
 
+class Vectorizer(TransformerMixin):
+    """Transforms n-dimensional array into 2D array of n_sample by n_features.
+
+    This class reshapes an n-dimensional array into an n_sample * n_features
+    array, usable by the estimators and transformers of scikit-learn.
+
+    Examples
+    --------
+    clf = make_pipeline(SpatialFilter(), XdawnTransformer(), Vectorizer(),
+                        LogisticRegression())
+
+    Attributes
+    ----------
+    ``shape_`` : array
+         Stores the original dimension of data.
+    """
+
+    def fit(self, X, y=None):
+        """Stores the shape of X.
+
+        Parameters
+        ----------
+        X : array-like
+            The data to fit. Can be, for example a list, or an array of at
+            least 2d. The first dimension must be of shape (n_samples).
+        y : None
+            Used for scikit-learn compatibility.
+
+        Returns
+        -------
+        self : Instance of Vectorizer
+            Return the modified instance.
+        """
+        self.shape_ = X.shape[1:]
+        return self
+
+    def transform(self, X):
+        """Convert matrix data into two dimensions.
+
+        Parameters
+        ----------
+        X : array-like
+            The data to be transformed. Can be, for example a list, or an
+            array of at least 2d. The first dimension must be of
+            shape (n_samples).
```

Comment:
The first dimension must be of shape (n_samples).
->
The first dimension must be of length n_samples.

fix below too